backend/core/MegaParse/megaparse/Converter.py

```diff
@@ -320,22 +320,24 @@ async def convert(
         else:
             raise ValueError(f"Method {self.method} not supported")
 
-        if not gpt4o_cleaner:
-            return LangChainDocument(
-                page_content=parsed_md,
-                metadata={"filename": file_path.name, "type": "pdf"},
-            )
-        else:
+        if gpt4o_cleaner:
             md_processor = MarkdownProcessor(
                 parsed_md,
                 strict=True,
                 remove_pagination=True,
             )
             md_cleaned = md_processor.process(gpt4o_cleaner=gpt4o_cleaner)
-            return LangChainDocument(
-                page_content=md_cleaned,
-                metadata={"filename": file_path.name, "type": "pdf"},
-            )
+            parsed_md = md_cleaned
+
+        if len(parsed_md) < 5 and self.strategy == "fast":
+            logger.debug(f"Switching to auto strategy for {file_path.name}")
+            self.strategy = "auto"
```

Comment:
This will set the strategy to "auto" for all subsequent parse if we use the same object 🤔